### PR TITLE
Fixes load of selectWoo and script in customizer to align the dropdow…

### PR DESCRIPTION
**Pull request by luislard**
…ns with the styles

### Describe the bug

tl;dr; Dropdowns from widgets inconsistent in Customizer and front-end

When using a standard "Filter by" widget in the sidebar and choosing to display that as a dropdown, the way it looks in the Customizer is different than the way it looks on the site.

## To Reproduce
1. Go to the Customizer → Widgets → Sidebar
2. Add a widget that can show dropdowns (e.g. "Filter products by Attribute" or "Product Categories")
3. Make sure "Show as dropdown" is checked
4. Publish
5. Observe the way the dropdown looks in the Customizer
6. Compare to the way it looks on the site

## Screenshots
Customizer:
dropdown-customizer

Front end:
site-dropdown

## Expected behavior
Looks should be consistent in Customizer and front end.

### All Submissions:

* [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/galleria/issues/99 .

### How to test the changes in this Pull Request:

1. Create a page and open customizer
2. Add a Product_cat Widget
3. in the widget select the option show as dropdown.

To test the filter product by attributes
1. Create a shop page
2. Open the customizer uin the shop page
3. Add the widget  filter product by attributes

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Loaded selectWoo in document inside customizer
> Throw an event when the lib is available
> Listen to the event on the widgets to properly initialize them

### CODE REVIEW

**Comment by jorgeatorres** [comment]
Hi @luislard!

First of all, apologies for the delay in getting back to you and let me thank you for your contribution! 💯 It definitely makes things more consistent.

In order to speed things up, I've rebased your changes off `trunk` to fix any conflicts.
That said, it's been a while since you first worked on this PR, so please let me know if you're still interested in this.

If so, there are a few things to address before we move forward:

- The code introduced has some tiny code standards violations, which you can [find here](https://github.com/woocommerce/woocommerce/runs/6917867598?check_suite_focus=true) in the logs from the CI run.
- A few of the widgets you're targeting here (for instance, the "Product Categories" one) are no longer available and have been replaced by blocks equivalents, which as far as I've seen, no longer use selectWoo/select2. A bit of research might be needed to see if this is still relevant or if we should be contributing the changes directly to the [woocommerce/woocommerce-blocks](https://github.com/woocommerce/woocommerce-blocks) repo.

Please let us know if you have any questions. Thanks again!

**Comment by jorgeatorres** [comment]
Closing due to inactivity. Feel free to re-open if you address the feedback above and want to get this reviewed again.
